cobra/flux_analysis/deletion.py

```diff
@@ -0,0 +1,348 @@
+from itertools import product
+from operator import attrgetter
+from collections import defaultdict
+from builtins import (map, dict)
+from future.utils import raise_
+
+from ..manipulation.delete import (find_gene_knockout_reactions)
+import cobra.util.solver as sutil
+
+try:
+    import scipy
+except ImportError:
+    moma = None
+else:
+    from . import moma
+
+try:
+    from pandas import DataFrame
+except:
+    DataFrame = None
+
+LOGGER = logging.getLogger(__name__)
+
+
+def infinite_defaultdict():
+    return defaultdict(infinite_defaultdict)
+
+
+def defaultdict_to_dict(def_dict):
+    if not isinstance(def_dict, defaultdict):
+        return def_dict
+    for k, v in def_dict.items():
+        def_dict[k] = defaultdict_to_dict(v)
+    return dict(def_dict)
+
+
+def _reactions_knockouts_with_restore(model, reactions):
+    to_restore = {}
+    growth = None
+    for reaction in reactions:
+        to_restore[reaction] = reaction.bounds
+        reaction.knock_out()
+    try:
+        solution = model.optimize()
+        if solution.status != 'infeasible':
+            growth = solution.objective_value
+    except:
+        pass
+    for reaction in reactions:
+        reaction.bounds = to_restore[reaction]
+    return (growth, [r.id for r in reactions])
+
+
+def _reaction_deletion_worker(ids):
+    model = _reaction_deletion_worker.model
+    return _reactions_knockouts_with_restore(
+        model,
+        [model.reactions.get_by_id(r_id) for r_id in ids]
+    )
+
+
+def _gene_deletion_worker(ids):
+    model = _gene_deletion_worker.model
+    all_reactions = []
+    for g_id in ids:
+        all_reactions.extend(find_gene_knockout_reactions(model, (model.genes.get_by_id(g_id),)))
+    growth, _ = _reactions_knockouts_with_restore(model, all_reactions)
+    return (growth, ids)
+
+
+def _multi_deletion(cobra_model, entity, element_lists, method="fba",
+                    num_cpu=None, solver=None, zero_cutoff=1e-12, **kwargs):
+    """
+    Helper function that provides the common interface for sequential knockouts
+
+    Parameters
+    ----------
+    cobra_model : cobra.Model
+        The metabolic model to perform deletions in.
+
+    element_lists : list
+        List of lists `cobra.Reaction`s or `cobra.Gene`s (or their IDs)
+        to be deleted.
+
+    method: {"fba", "moma"}, optional
+        Procedure used to predict the growth rate.
+
+    solver: str, optional
+        This must be a QP-capable solver for MOMA. If left unspecified,
+        a suitable solver will be automatically chosen.
+
+    zero_cutoff: float, optional
+        When checking to see if a value is 0, this threshold is used.
+
+    Returns
+    -------
+    dict of dict
+        A sparse representation of all combinations of pairs of reaction
+        deletions without replacements.
+    """
+    with cobra_model as model:
+        WORKERS = dict(gene=_gene_deletion_worker, reaction=_reaction_deletion_worker)
+        worker_function = WORKERS[entity]
```

Comment:
I would skip assigning the `dict` to anything. i.e., `worker_function = dict(...)[entity]`.